extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -18,6 +18,12 @@ import {ArrowNext, ArrowPrev} from './arrow';
 import {Scroller} from './scroller';
 import {toChildArray, useRef, useState} from '../../../src/preact';
 import {useMountEffect} from '../../../src/preact/utils';
+import styles from './base-carousel.jss';
+import {createUseStyles} from 'react-jss';
+
+// TODO discuss this sidestepping of local/no-export-side-effect.
+// eslint-disable-next-line local/no-export-side-effect
+export const useStyles = createUseStyles(styles);
```

Comment:
I liked your idea of putting this into the `base-carousel.jss.js` because:
1. It removes the danger of misusing the style objects.
2. It makes it really obvious where to find `useStyles` styles.
3. Limits the `react-jss` import to only one type of module.
4. Simplifies rewriting conceptually - only one type of module is rewritten. It could also make rewriting slightly more difficult technically, but I think it's a good tradeoff.
5. Also, down the line I'd like a lint that only allows `react-jss` and `jss` to be imported in a `.jss.js` modules.